p2p/libp2p/discovery/kadDhtDiscoverer.go

```diff
@@ -229,3 +237,33 @@ func (kdd *KadDhtDiscoverer) IsInterfaceNil() bool {
 	}
 	return false
 }
+
+// StartWatchdog start the watchdog, and return the kick channel
+func (kdd *KadDhtDiscoverer) StartWatchdog(timeout time.Duration) chan struct{} {
+	kdd.mutKadDht.Lock()
+	defer kdd.mutKadDht.Unlock()
+
+	if kdd.contextProvider == nil {
+		return nil
+	}
+
+	if kdd.watchdogKick != nil {
+		return kdd.watchdogKick
```

Comment:
This should yield an error as the watchdog process is already started. Otherwise, the function does at least 2 things: start a watchdog and, if already started, returns the same chan.

Reply:
Done, (changed the interface)